### PR TITLE
[incubator-kie-issues#1068] Fix gradle build output

### LINKED ISSUE #1068: Fix gradle build output
During codegen, the AppPaths verify the building system (maven vs gradle) and adapt directorys/paths based on that - namely gradle output directory is "build" , and maven is "target".
Unfortunately, elsewhere the "target" output is hardcoded, leading to mix of maven and gradle defaults.
Worse, in case of gradle, the output directory is created inside the build one, with a nested directory effect:

Image

### PR DESCRIPTION
**Pull request by gitgabrio**
Fixes https://github.com/apache/incubator-kie-issues/issues/1068

Needed by https://github.com/apache/incubator-kie-kogito-runtimes/pull/3471

This PR:
1. expand the scope of the AppPAths.BuildTool
2. replace all usages of hardoded directories/paths outside it
3. enforce the GRADLE-related output paths

Note:
Quarkus define build/classes/java/main as classes dir (vs targer/classes), so this could be a problem with Springboot+gradle (currently untested - still in draft)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Review by baldimir — APPROVED** [review]

**Review by porcelli — APPROVED** [review]
lgtm

**Review by mariofusco — APPROVED** [review]

File: drools-model/drools-codegen-common/src/test/java/org/drools/codegen/common/GeneratedFileWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package org.drools.codegen.common;
```
Comment (yesamer) [inline drools-model/drools-codegen-common/src/test/java/org/drools/codegen/common/GeneratedFileWriterTest.java:19]: @gitgabrio Please add the Copyright/license Apache header

File: drools-model/drools-codegen-common/src/main/java/org/drools/codegen/common/GeneratedFileWriter.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/**
+/*
```
Comment (yesamer) [inline drools-model/drools-codegen-common/src/main/java/org/drools/codegen/common/GeneratedFileWriter.java]: Typo?

**Review by yesamer — APPROVED** [review]
Well done, @gitgabrio, very nice improvement.

**Pull request merged**